home2.1/task3.java

```diff
@@ -0,0 +1,95 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Scanner;
+//In this code, Scanner is used to read data from a file.
+
+public class task3 {
```

Comment:
You can apply almost all the comments from C++ version - we should encapsulate logic of work with history into a class, named constants should be used and so on.